### ISSUE TITLE
Unseeded np.random.default_rng() in _full_update.py breaks reproducibility of model initialization

### ISSUE THREAD

**Issue by jc-macdonald** [yoavram-lab/VBPCApy#109]
## Problem

`src/vbpca_py/_full_update.py:680`:

```python
rng=None if init_value else np.random.default_rng(),
```

When `init_value` isn't supplied, VBPCA's initialization draws from a **fresh, unseeded** `np.random.default_rng()` on every call. There is no `random_state`/`seed` parameter anywhere on `VBPCA.__init__` or `.fit()` to control this. Two `.fit()` calls with identical data, mask, and hyperparameters can therefore produce different results, and there is no way to make them reproducible.

## Why this matters

Found while auditing the hyperparameter trade study in `analysis/trade_study` (which produces `recommend_config`'s shipped defaults): the study's simulator (`_world.py`) seeds *data generation* deterministically per regime, but every `VBPCA.fit()` call downstream of that still has uncontrolled initialization noise. Combined with the trade study evaluating each design point only once (see companion issue on the trade-study side), this means the "scores" feeding the hyperparameter recommendation surrogate are noisier and less reproducible than they need to be — and this can't be fixed at the trade-study level without library support.

More broadly: any user trying to get bit-for-bit reproducible results, or comparing two hyperparameter configs fairly, currently can't, because part of the randomness is uncontrollable.

## Ask

Add a `random_state`/`seed` parameter to `VBPCA.__init__` (and thread it through to `.fit()`), used to seed the internal `np.random.default_rng()` call(s) in `_full_update.py` (and anywhere else internal randomness is used). Should follow the sklearn `random_state` convention given VBPCA's existing sklearn-compatible API.
